### PR TITLE
Added a clone function to BasisSet.

### PR DESCRIPTION
**Pull request by psavery**
Added a virtual copy constructor to BasisSet so that it may be copied.

### CODE REVIEW

**Comment by psavery** [comment]
Let me know what you think about this. I couldn't do a direct copy of BasisSet because it is an abstract class.

File: avogadro/core/basisset.h [hunk 1]
```diff
@@ -49,6 +49,11 @@ class AVOGADROCORE_EXPORT BasisSet
   virtual ~BasisSet() {}
 
   /**
+   * Virtual copy constructor
```
Comment (cryos) [inline avogadro/core/basisset.h]: It isn't a copy constructor, let's just call if a clone function.

File: avogadro/core/gaussianset.h [hunk 2]
```diff
@@ -66,6 +66,11 @@ class AVOGADROCORE_EXPORT GaussianSet : public BasisSet
   ~GaussianSet() AVO_OVERRIDE;
 
   /**
+   * Virtual copy constructor.
+   */
+  GaussianSet* clone() const AVO_OVERRIDE {return new GaussianSet(*this);};
```
Comment (cryos) [inline avogadro/core/gaussianset.h]: Don't need the final ;, please add a little white space such that

{ return new GaussianSet(*this); }

File: avogadro/core/slaterset.h [hunk 3]
```diff
@@ -62,6 +62,11 @@ class AVOGADROCORE_EXPORT SlaterSet : public BasisSet
   ~SlaterSet() AVO_OVERRIDE;
 
   /**
+   * Virtual copy constructor.
+   */
+  SlaterSet* clone() const AVO_OVERRIDE {return new SlaterSet(*this);};
```
Comment (cryos) [inline avogadro/core/slaterset.h]: Space additions, and remove extra semicolon as above

**Comment by psavery** [comment]
Will do. About to travel. Will update it later today.

**Comment by cryos** [comment]
No rush, all fairly minor things. I will be leaving in about a half hour and less available until Saturday.

**Comment by psavery** [comment]
Does that look fine now?

**Comment by cryos** [comment]
Looks good to me.

**Pull request merged**